python/ray/autoscaler/_private/kuberay/node_provider.py

```diff
@@ -202,6 +205,81 @@ def _worker_group_replicas(raycluster: Dict[str, Any], group_index: int):
     return raycluster["spec"]["workerGroupSpecs"][group_index].get("replicas", 1)
 
 
+class IKubernetesHttpApiClient(ABC):
```

Comment:
i can potentially move these changes to another PR.

Reply:
this is merely moving v1's codes around into a client class that's mockable for v2 provider testing.

Reply:
It's a nice refactor!